### PR TITLE
432/show vSphere path in VMs show

### LINKED ISSUE #432: Display vsphere folder path on vm show page
**Description**

The system should display the vsphere folder path on the vm show page in the section "Hardware / Network".

### LINKED ISSUE #438: Refactor custom ip validations for server model
**Description**

Replace the unreadable regular expression by using a gem: https://stackoverflow.com/questions/3634998/how-do-i-check-whether-a-value-in-a-string-is-an-ip-address

### LINKED ISSUE #477: Management & Caching of AppSettings
Currently, every `AppSettings.instance` call results in a db query... Those are cached, but maybe https://github.com/huacnlee/rails-settings-cached could help.

### PR DESCRIPTION
**Pull request by MrModder**
Closes #432, fixes #438 and also closes #477

### CODE REVIEW

File: app/views/vms/show.html.erb [hunk 1]
```diff
@@ -7,7 +7,13 @@
 end %>
 
 <div class="container mb-3 pl-0 pr-0 clearfix">
-  <h1 class="mb-0"><%= @vm.name unless @vm.nil? %></h1>
+  <ol class="breadcrumb" aria-label="The Virtual Machines full path">
```
Comment (BjoernDaase) [inline app/views/vms/show.html.erb]: I like the positioning on the page :+1:

File: app/models/app_setting.rb [hunk 2]
```diff
@@ -16,13 +16,19 @@ class AppSetting < ApplicationRecord
 
   after_commit :apply_settings, on: :update
 
+  def self.refresh_instance
+    @instance = nil
```
Comment (chrisma) [inline app/models/app_setting.rb]: Would `clear_cache` be a better name for this method?

File: app/models/server.rb [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
+# Part of Ruby stdlib, used for IP regexes
+# https://www.rubydoc.info/stdlib/resolv/Resolv
+require 'resolv'
```
Comment (chrisma) [inline app/models/server.rb]: Nice use of the Ruby stdlib :+1:

File: app/models/app_setting.rb (hunk 2 above)
Comment (chrisma) [reply app/models/app_setting.rb]: https://github.com/hpi-swt2/vm-portal/pull/481/commits/1713c798863ae4602c06abbacf66848ac00a7a7f

**Pull request merged**